src/Engine/GameEvents/MoveEvent.java

```diff
@@ -44,34 +43,30 @@ public void performEvent(MapMaster gameMap)
     if( unitPath.getPathLength() > 0 ) // Make sure we have a destination.
     {
       Path.PathNode endpoint = unitPath.getEnd();
-      Location loc = gameMap.getLocation(endpoint.x, endpoint.y);
-      int fuelBurn = unitPath.getFuelCost(unit.model, gameMap);
+      int fuelBurn = unitPath.getFuelCost(unit, gameMap);
+      boolean includeOccupied = false;
 
-      // If the unit is already at the destination, we don't need to move it.
-      if( 0 == fuelBurn )
-      {
-        unit.isTurnOver = true;
-      }
-      // Make sure it is valid to move this unit to its destination.
-      else if( loc.getResident() == null && unit.model.propulsion.getMoveCost(loc.getEnvironment()) < 99 )
-      {
-        gameMap.moveUnit(unit, endpoint.x, endpoint.y);
+      unit.fuel = Math.max(0, unit.fuel - fuelBurn);
 
-        unit.fuel -= fuelBurn;
+      // Every unit action begins with a (potentially 0-distance) move,
+      // so we'll just set the "has moved" flag here.
+      unit.isTurnOver = true;
 
-        // Every unit action begins with a (potentially 0-distance) move,
-        // so we'll just set the "has moved" flag here.
-        unit.isTurnOver = true;
-
-        // reveal fog as applicable
-        for( Commander co : gameMap.commanders )
-        {
-          co.myView.revealFog(unit, unitPath);
-        }
+      // reveal fog as applicable
+      for( Commander co : gameMap.commanders )
+      {
+        co.myView.revealFog(unit, unitPath);
```

Comment:
Shouldn't we wait until we know where the unit is moving before clearing fog?

Reply:
We already have the full `unitPath`, though?
If you're talking about preemption, that's done during event construction.

Reply:
Yes, and based on the pre-PR version of this function, the event should be ready to fire. You added a new check below to ensure `fuelBurn <= unit.model.movePower && etc.` before actually executing the move.
Either we need that check (in which case we should wait to clear fog until we know for sure we are moving) or we don't need that check.

Reply:
I believe the move execution function does validate that you can do the actual move too, so I don't think the check is actually needed -- there are safeguards earlier up the chain that should prevent bad things from happening.
I was optimizing for the case (as detailed below) in which we fire a `MoveEvent` onto an occupied space (i.e. Load or Join). Logically, the fuel cost/end turn/fog reveal *should* still occur in that case.

Reply:
Well, then--and write this down, because I don't say this often--I think you should remove the check so that it doesn't look like our logic is inconsistent and cause questions when we look at the function in six months.